### PR TITLE
add fides package

### PR DESCRIPTION
**Pull request by cyrush**

### CODE REVIEW

File: var/spack/repos/builtin/packages/fides/package.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class Fides(CMakePackage) :
+    """A library that provides a schema for ADIOS2 streams.
+    """
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/fides/package.py]: Optional, but triggers my OCD:
```suggestion
    """A library that provides a schema for ADIOS2 streams."""
```

File: var/spack/repos/builtin/packages/fides/package.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    homepage = "https://gitlab.kitware.com/vtk/fides"
+    url      = "https://gitlab.kitware.com/vtk/fides/-/archive/v1.0.0/fides-v1.0.0.tar.gz"
+    git      = "https://gitlab.kitware.com/vtk/fides.git"
+
+    maintainers = ['caitlin.ross', 'dpugmire']
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/fides/package.py]: The first maintainer doesn't seem to be a valid Github username
```suggestion
    maintainers = ['dpugmire']
```

File: var/spack/repos/builtin/packages/fides/package.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    depends_on("vtk-m", when="+vtk-m")
+
+    def cmake_args(self):
+        spec = self.spec
+
+        options = []
+        options.append("-DVTKm_DIR={0}".format(spec['vtk-m'].prefix))
+        options.append("-DADIOS2_DIR={0}".format(spec['adios2'].prefix))
+        options.append("-DFIDES_ENABLE_TESTING=OFF")
+        options.append("-DFIDES_ENABLE_EXAMPLES=OFF")
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/fides/package.py]: You can use the more idiomatic:
```suggestion
        options = [
            self.define("VTKm_DIR", spec['vtk-m'].prefix),
            self.define("ADIOS2_DIR", spec['adios2'].prefix),
            self.define("FIDES_ENABLE_TESTING", "OFF"),            
            self.define("FIDES_ENABLE_EXAMPLES", "OFF")
        ]
```
Beware I wrote the code during review and never tried it (so there might be typos).

**Review by alalazo — CHANGES_REQUESTED** [review]
A few minor points and `style` test is failing

**Comment by cyrush** [comment]
@alalazo thanks! I fixed the github name, made and tested the other suggestions.

**Review by alalazo — APPROVED** [review]

**Comment by cyrush** [comment]
@alalazo thanks!

**Pull request merged**